File: LeetCode/2341. Maximum Number of Pairs in Array/solution.py
"""
Runtime: 40 ms
Memory Usage: 15 MB
"""
class Solution:
    def numberOfPairs(self, nums: List[int]) -> List[int]:
        pair = sum(c // 2 for c in Counter(nums).values())
        return [pair, len(nums) - pair * 2]

File: LeetCode/2341. Maximum Number of Pairs in Array/solution.cpp
/**
 * Runtime: 4 ms
 * Memory Usage: 8.9 MB 
 */
class Solution {
public:
    vector<int> numberOfPairs(vector<int>& nums) {
        unordered_map<int, int> cnt;
        for (auto &num: nums) {
            cnt[num] += 1;
        }
        int pair = 0;
        for (auto &[_, c]: cnt) {
            pair += c / 2;
        }
        return {pair, int(nums.size()) - pair * 2};
    }
};
